cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/SparkStreamingPipelineRunner.java

```diff
@@ -49,42 +59,70 @@
 import io.cdap.cdap.etl.spark.streaming.function.CountingTransformFunction;
 import io.cdap.cdap.etl.spark.streaming.function.DynamicJoinMerge;
 import io.cdap.cdap.etl.spark.streaming.function.DynamicJoinOn;
+import io.cdap.cdap.etl.spark.streaming.function.StreamingBatchSinkFunction;
+import io.cdap.cdap.etl.spark.streaming.function.StreamingMultiSinkFunction;
+import io.cdap.cdap.etl.spark.streaming.function.StreamingSparkSinkFunction;
 import io.cdap.cdap.etl.spark.streaming.function.WrapOutputTransformFunction;
 import io.cdap.cdap.etl.spark.streaming.function.preview.LimitingFunction;
 import io.cdap.cdap.etl.validation.LoggingFailureCollector;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.sql.SQLContext;
+import org.apache.spark.streaming.Time;
 import org.apache.spark.streaming.api.java.JavaDStream;
 import org.apache.spark.streaming.api.java.JavaPairDStream;
 import org.apache.spark.streaming.api.java.JavaStreamingContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import javax.annotation.Nullable;
 
 /**
  * Driver for running pipelines using Spark Streaming.
  */
 public class SparkStreamingPipelineRunner extends SparkPipelineRunner {
 
+  private static final Logger LOG = LoggerFactory.getLogger(SparkStreamingPipelineRunner.class);
+
   private final JavaSparkExecutionContext sec;
-  private final JavaStreamingContext streamingContext;
+  private final JavaStreamingContext javaStreamingContext;
   private final DataStreamsPipelineSpec spec;
-  private final boolean checkpointsDisabled;
+  private final boolean nativeStateStoreEnabled;
```

Comment:
Be careful with the word `native`, as it can mean many things in Java. Do you mean `useStateStore` as oppose to the Spark checkpoint?

Reply:
Removed native from the name to avoid confusion